src/log/log.c

```diff
@@ -1215,7 +1215,7 @@ __log_newfile(WT_SESSION_IMPL *session, bool conn_open, bool *created)
          * Increment the missed pre-allocated file counter only if a hot backup is not in progress.
          * We are deliberately not using pre-allocated log files during backup (see comment above).
          */
-        if (conn->hot_backup_start == 0)
+        if (conn->hot_backup_start == 0 && conn_open == false)
```

Comment:
I do not think this is the correct fix. It becomes dead code after `wiredtiger_open` completes and that is not the intention. The system being able to react and adjust during running time is more important than something at startup. With your change, a running system will never take this conditional. 

Please remove this part of the change.

Reply:
I re-reviewed this again in the code itself (instead of just the change). I understand what you're doing now. I have two minor comments and then I think this will be ready.
1. Our style prefers `!conn_open` instead of comparing with false. 
2. Can you modify the comment above, at line 1215, to say: 
```
Increment the missed pre-allocated file counter only during normal operations and when a hot backup
is not in progress. We are deliberately ... 
```